src/module.cc

```diff
@@ -0,0 +1,338 @@
+#include <nan.h>
+//#include <node.h>
+#include "node_report.h"
+
+// Internal/static function declarations
+static void OnFatalError(const char* location, const char* message);
+bool OnUncaughtException(v8::Isolate* isolate);
+#ifndef _WIN32
+static void SignalDumpAsyncCallback(uv_async_t* handle);
+inline void* ReportSignalThreadMain(void* unused);
+static int StartWatchdogThread(void *(*thread_main) (void* unused));
+static void RegisterSignalHandler(int signal, void (*handler)(int signal), bool reset_handler);
+static void SignalDump(int signo);
+#endif
+
+// Default nodereport option settings
+static unsigned int nodereport_events = NR_EXCEPTION + NR_FATALERROR + NR_SIGNAL + NR_JSAPICALL;
+static unsigned int nodereport_core = 1;
+static unsigned int nodereport_verbose = 0;
+
+#ifdef _WIN32
+static unsigned int nodereport_signal = 0; // no-op on Windows
+#else  // signal support - on Unix/OSX only
+static unsigned int nodereport_signal = SIGUSR2; // default signal is SIGUSR2
+static int report_signal = 0;
+static uv_sem_t report_semaphore;
+static uv_async_t nodereport_trigger_async;
+static uv_mutex_t node_isolate_mutex;
+#endif
+
+static v8::Isolate* node_isolate;
+
+/*******************************************************************************
+ * External JavaScript API for triggering a NodeReport
+ *
+ ******************************************************************************/
+NAN_METHOD(TriggerReport) {
+  Nan::HandleScope scope;
+  v8::Isolate* isolate = Isolate::GetCurrent();
+  char filename[48] = "";
+
+  if (info[0]->IsString()) {
+    // Filename parameter supplied
+    Nan::Utf8String filename_parameter(info[0]->ToString());
+    if (filename_parameter.length() < 48) {
+      strcpy(filename, *filename_parameter);
+    } else {
+      Nan::ThrowSyntaxError("TriggerReport: filename parameter is too long (max 48 characters)");
+    }
+  }

```

Comment:
Use a std::string so you don't have to impose a fairly arbitrary size limit?  At the very least compare against `sizeof(filename)` instead of a hard-coded value.